app/src/main/java/com/github/alfabravo2013/readyforexams/presentation/home/ChecklistAdapter.kt

```diff
@@ -0,0 +1,84 @@
+package com.github.alfabravo2013.readyforexams.presentation.home
+
+import android.content.res.ColorStateList
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.core.widget.ImageViewCompat
+import androidx.recyclerview.widget.RecyclerView
+import com.github.alfabravo2013.readyforexams.R
+import com.github.alfabravo2013.readyforexams.databinding.ItemHomeChecklistBinding
+import com.github.alfabravo2013.readyforexams.domain.models.Checklist
+
+class ChecklistAdapter : RecyclerView.Adapter<ChecklistAdapter.ViewHolder>() {
+    private val checklists = mutableListOf<Checklist>()
+
+    class ViewHolder(
+        private val binding: ItemHomeChecklistBinding
+    ) : RecyclerView.ViewHolder(binding.root) {
+        private var statusColor = 0
+        private lateinit var statusText: String
+        private lateinit var tasksCountText: String
+
+        fun bind(item: Checklist) {
+            with(binding) {
+                setVariables(item.completed, item.total)
+
+                homeChecklistItemName.text = item.name
+                homeChecklistItemTasksCount.text = tasksCountText
+                homeChecklistItemStatusText.text = statusText
+
+                ImageViewCompat.setImageTintList(
+                    homeChecklistItemStatusIcon,
+                    ColorStateList.valueOf(statusColor)
+                )
+
+                homeChecklistItemStatusText.setTextColor(statusColor)
+            }
+        }
+
+        private fun setVariables(completed: Int, total: Int) {
+            val context = binding.root.context
+
+            tasksCountText =
+                context.getString(R.string.home_checklist_tasks_count_text, completed, total)
+
+            when (completed) {
```

Comment:
There should be no logic inside the adapter also. You can ignore this comment for now and we will move back to it a bit later.